sktime/forecasting/compose/_pipeline.py

```diff
@@ -1583,10 +1583,14 @@ def _get_X_for_fcX(self, X):
             return None
 
         if ixx == "complement":
-            X_for_fcX = X.drop(columns=self.columns)
-            if X_for_fcX.shape[1] < 1:
+            if X is None:
                 return None
-            return X_for_fcX
+            all_columns = X.columns
+            if self.columns is not None:
```

Comment:
Related to above comment, this check is probably redundant.

Reply:
removing it